### ISSUE TITLE
Problem with rendering greek (mathematical) character when in sequence with another normal character.

### ISSUE THREAD

**Issue by denis97** [wkhtmltopdf/wkhtmltopdf#4837]
**wkhtmltopdf version(s) affected**: 0.12.6

**OS information**
Windows 10

**Description**
When converting an html with fonts binded to css I get a diverse font used for greek characters when followed by other characters. If the character is alone in the html <span> it gets rendered with the right font

**How to reproduce**

`<span class="s1 f11 fs9 l0" id="P21_541">π</span>` is rendered right while
`<span class="s1 f11 fs9 l0" id="P21_454">πw</span>` have "pi" rendered with a different font( Tahoma) and w is still right

**Expected behavior**

**Possible Solution**

**Comment by PhilterPaper** [comment]
In what order is CSS specifying the font-family? Do all of these fonts have 'w', and which ones have pi?  Is there any difference if your text is w _followed_ by pi?

**Comment by denis97** [comment]
there's no difference if i switch the order. The fonts have all the characters used
[testt.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/5276972/testt.zip)

**Comment by denis97** [comment]
> In what order is CSS specifying the font-family? Do all of these fonts have 'w', and which ones have pi? Is there any difference if your text is w _followed_ by pi?

**Comment by PhilterPaper** [comment]
Please supply a minimal HTML test case to show the problem. I can't even get it to properly show in a browser when I use your CSS. It should, of course, work properly in a browser before I attempt to run wkHTMLtoPDF.

With your CSS, the first "pi" doesn't even show up, while the pi-w is enormous. I can see that the "pi" is double-struck to fake bold text, while the "w" is native bold font. Maybe that's a clue to what's going on?

**Comment by denis97** [comment]
Sorry, i share an entire page with you. I think that maybe it's dimension related and not to order or anything
[test2.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/5277678/test2.zip)

**Comment by PhilterPaper** [comment]
OK, it looks fine in a browser, but there is a difference in the formula in the line just before "Rayleigh range". The pi turns from a lowercase normal weight 'π' to a **bold** CAPITAL Pi. Is that what you're seeing? I can't see anything in the CSS that would be uppercasing and bolding a Greek letter, so it must be a bug in wkHTMLtoPDF. I dumped the f11.otf font file and see only 79 glyphs in it, including space <>.,/ 25 uppercase Latin (no Q), 26 lowercase Latin, and 22 lowercase Greek (including π). Possibly some Latin lowercase characters do double duty as Greek lowercase? Unless I missed it, that is the only use of π on the page. Anyway, both π and _w_ are in the f11.otf font, so I wouldn't think it would do a substitution from another font.

By the way, in equation 1.8, I don't see the horizontal divisor lines (two of them) until I scale up to 125% or larger. Do you see that? It seems odd, but it may just be a scaling thing in the PDF reader.

**Comment by denis97** [comment]
Yes, π becomes bold but also changes font in the pdf... it is tahoma font by analyzing it. The main problem is that is repeated for a lot of greek chracters like π in other pages...

**Comment by PhilterPaper** [comment]
Well, it must have changed font, as there is no capital 'Pi' in the f11.otf font. I won't say for sure that it's bold, but it sure looks that way. When I tried it before you sent the full page, it was bolding the π by double-striking it (with offset). It looks like it is still bolding it for some reason. Maybe wkHTMLtoPDF got confused and uppercased π for some reason, and decided it had to go to a different font (Tahoma) to find this character (while leaving _w_ in the correct font?). The bold may have been introduced in the same mistake.

Anyway, that's all I can see at this point. It smells like a bug in wkHTMLtoPDF (I'm using 0.12.5 with patched qt on Windows 10), which is a shame, as other than that it seems to have done a decent job of conversion (the .xhtml format helped by specifying the exact location, font, and size of each word or smaller item). Unless there is some subtle CSS feature that's breaking it, I don't think we need to round up the usual suspects (HTML using more advanced JS and CSS than wkHTMLtoPDF supports).
